base/src/TestSignalGeneratorSrc.cpp

```diff
@@ -106,4 +95,16 @@ void TestSignalGenerator::setMetadata(framemetadata_sp &metadata)
     {
         return;
     }
-}
\ No newline at end of file
+}
+
+TestSignalGeneratorProps TestSignalGenerator::getProps()
+{
+    return mDetail->mProps;
+}
+
+void TestSignalGenerator::setProps(TestSignalGeneratorProps& _props)
+{
+    mDetail->mProps = _props;
```

Comment:
better to have a Detail::setProps method that is called from here. We might want to do more things in Detail based on props change. Also, the read_size calculation should happen here.